coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -265,9 +267,13 @@ private SqlNode visitLateralViewUDTF(List<SqlNode> sqlNodes, List<SqlNode> alias
   private SqlNode visitLateralViewExplode(List<SqlNode> sqlNodes, List<SqlNode> aliasOperands,
       SqlCall tableFunctionCall, boolean isOuter) {
     final int operandCount = aliasOperands.size();
-    // array if operandCount == 3: LATERAL VIEW EXPLODE(op0) AS op1(op2)
-    // map if operandCount == 4: LATERAL VIEW EXPLODE(op0) AS op1(op2, op3)
-    checkState(operandCount == 3 || operandCount == 4,
+    // array if operandCount == 3: LATERAL VIEW EXPLODE(op0) op1 AS op2
+    // map if operandCount == 4: LATERAL VIEW EXPLODE(op0) op1 AS op2, op3
+    // if operandCount == 2: LATERAL VIEW EXPLODE(op0) op1
+    //   if op0 is a map, it implies "AS key, value" where key, value are auto-named by Hive
+    //   if op0 is an array, is implies "AS col" where col is auto-named by Hive
+    //   The logic above will be implemented as part of Calcite SqlNode validation
+    checkState(operandCount == 2 || operandCount == 3 || operandCount == 4,
```

Comment:
For the `operandCount == 2` case, I think we also need to add the support if `isOuter == true`.
i.e. 
```
SELECT a_alias.col FROM (SELECT ARRAY('a1', 'a2') as a) tmp LATERAL VIEW OUTER EXPLODE(a) a_alias
```

Reply:
You are right that `operandCount == 2` `OUTER` case is not supported right now.

I couldn't think about a way to support it easily since we don't know the type of the parameter at this point.

Let me throw an exception and add a test case to catch this.